ingester/user_state.go

```diff
@@ -28,10 +29,11 @@ type userState struct {
 	ingestedSamples *ewmaRate
 }
 
-func newUserStates(rateUpdatePeriod time.Duration) *userStates {
+func newUserStates(rateUpdatePeriod time.Duration, maxSeriesPerUser int) *userStates {
```

Comment:
I'd probably propagated the whole of Config struct here, as we're already passing two things through.  Or better yet, make a new Config struct (`UserLimitsConfig`), embed that in Config and pass that here?